server/src/main/java/org/elasticsearch/common/settings/SecureSetting.java

```diff
@@ -97,6 +97,23 @@ public T get(Settings settings) {
         }
     }
 
+    /**
+     * Returns the digest of this secure setting's value. This method can be called at all times, there is no requirement that
+     * {@code SecureSettings} be open, unlike {@code #get(Settings)}. The digest is used to check for changes of the value, without
+     * actually storing the value.
+     */
+    public byte[] getSecretDigest(Settings settings) {
+        final SecureSettings secureSettings = settings.getSecureSettings();
+        if (secureSettings == null || false == secureSettings.getSettingNames().contains(getKey())) {
+            return null;
+        }
+        try {
```

Comment:
What happens if we fallback to a default value? This would not exist in the keystore, and thus have no precomputed hash.

Reply:
Retrieving the hash for a secure setting does not follow the fallback.
All settings that rely on fallbacks will have a *null* hash. I think it is okay in the *consistency check* use case to only check consistency of the keystore, and close the eyes for different values as fallbacks.